assets/src/scripts/components/graph/view/construction.js

```diff
@@ -128,6 +131,55 @@ const drawPatterns = function (elem) {
         });
 };
 
+const drawWellBorehole =  function drawWellHole(store, container, opts, elements) {
+    // get the sate from the store
+    const state = store.getState();
+    // get the measured depth of the well
+    const rawDepth    = getSiteWellDepth(opts.agencyCode, opts.siteId)(state);
+    // scale the depth to the graph size
+    const scaleDepth  = getScaleY(opts, 'construction')(state)(rawDepth);
+    // check to see if we have construction elements
+    const hasElements = elements && elements.length > 0;
+    // if there are no elements then use max size of the graph, else use 0 for max function to work
+    const minDepth    = hasElements ? 520 : 0;
+    // if there is good depth info then use it else use a high value for min function to work
+    const maxDepth    = scaleDepth > 300 ? scaleDepth : 520;
+
+    // if there are elements use them to discover left and right otherwise use defaults
+    const left   = hasElements ? 100 : 8;
+    const right  = hasElements ? 8 : 100;
+
+    const borehole = {
+        title: 'borehole',
+        type: 'borehole',
+        thickness: 1,
+        left: {
+            x:  left,
+            y1: minDepth,
+            y2: maxDepth
+        },
+        right: {
+            x:  right,
+            y1: minDepth,
+            y2: maxDepth
+        }
+    };
+    elements.forEach((element) => {
+        borehole.left.x = Math.min(borehole.left.x, element.left.x);
+        borehole.right.x = Math.max(borehole.right.x, element.right.x);
+
+        borehole.left.y1 = Math.min(borehole.left.y1, element.left.y1);
+        borehole.right.y1 = Math.min(borehole.right.y1, element.right.y1);
+
+        borehole.left.y2 = Math.max(borehole.left.y2, element.left.y2);
+        borehole.right.y2 = Math.max(borehole.right.y2, element.right.y2);
+    });
```

Comment:
Git hub conversation management is not allowing me to continue in the right thread so I will add my comment here.

I do not get how I am supposed to refactor my drawBorehole when drawElement requires the store. Are you asking to refactor the drawElement too? I might have sounded obtuse because it feels outside the scope of the story and I was not expecting that suggestion. Should we create a story for this improvement? 

What are we missing in the conversation here? Do we need to have a call to hash this out? (Text is so incomplete it's frustrating.)